### ISSUE TITLE
spring-aot-maven-plugin fails on Windows

### ISSUE THREAD

**Issue by leccelecce** [spring-projects-experimental/spring-native#507]
Using 0.9.0-SNAPSHOT

Running `mvn clean install -U -DskipTests=true` I get the following:

```
[INFO] --- spring-aot-maven-plugin:0.9.0-SNAPSHOT:generate (generate) @ proj---
[INFO] Spring Native operating mode: native
[ERROR] org.springframework.nativex.type.MissingTypeException: Unable to find class file for com/ecv/webapp/Application
[ERROR] [org.springframework.nativex.type.TypeSystem.resolveSlashed(TypeSystem.java:217), org.springframework.nativex.type.TypeSystem.resolveSlashed(TypeSystem.java:178), org.springframework.nativex.type.TypeSystem.getStereoTypesOnType(TypeSystem.java:1180), ...
[INFO] No sources to compile
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory D:\GitProjects\proj\target\generated-sources\spring-aot\src\main\resources
```

If I then try to run my JAR file or debug my app in my IDE, I get the following failure:

```
Caused by: java.lang.IllegalStateException: Mandatory generated class org.springframework.aot.StaticSpringFactories not found, please make sure spring-aot-maven-plugin or spring-aot-gradle-plugin are configured properly.
        at org.springframework.nativex.NativeListener.<clinit>(NativeListener.java:32)
```

**Comment by leccelecce** [comment]
Similar issue also blocks my tests running:

```
[INFO] --- spring-aot-maven-plugin:0.9.0-SNAPSHOT:test-generate (test-generate) @ netballfantasyleague ---
[INFO] Spring Native operating mode: native
[ERROR] org.springframework.nativex.type.MissingTypeException: Unable to find class file for com/ecv/webapp/HttpServerTest
[ERROR] [org.springframework.nativex.type.TypeSystem.resolveSlashed(TypeSystem.java:217), org.springframework.nativex.type.TypeSystem.resolveSlashed(TypeSystem.java:178), org.springframework.nativex.type.TypeSystem.getStereoTypesOnType(TypeSystem.java:1180), java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195), java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195), java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625), java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658), java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274), java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625), java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484), java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474), java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913), java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234), java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578), org.springframework.nativex.type.TypeSystem.scanForSpringComponents(TypeSystem.java:1189), org.springframework.nativex.support.ResourcesHandler.synthesizeSpringComponents(ResourcesHandler.java:227), org.springframework.nativex.support.ResourcesHandler.handleSpringComponents(ResourcesHandler.java:216), org.springframework.nativex.support.ResourcesHandler.register(ResourcesHandler.java:105), org.springframework.nativex.support.SpringAnalyzer.analyze(SpringAnalyzer.java:73), org.springframework.aot.nativex.ConfigurationContributor.contribute(ConfigurationContributor.java:50), org.springframework.aot.BootstrapCodeGenerator.generate(BootstrapCodeGenerator.java:69), org.springframework.aot.maven.TestGenerateMojo.execute(TestGenerateMojo.java:44), org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137), org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210), org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156), org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148), org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117), org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81), org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56), org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128), org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305), org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192), org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105), org.apache.maven.cli.MavenCli.execute(MavenCli.java:957), org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289), org.apache.maven.cli.MavenCli.main(MavenCli.java:193), java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method), java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64), java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43), java.base/java.lang.reflect.Method.invoke(Method.java:564), org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282), org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225), org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406), org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347), org.codehaus.classworlds.Launcher.main(Launcher.java:47)]
[INFO] No sources to compile
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory D:\GitProjects\fantasyleague\target\generated-test-sources\spring-aot\src\test\resources
```

**Comment by sdeleuze** [comment]
Please share a repro project.

**Comment by leccelecce** [comment]
[repro-sn507.zip](https://github.com/spring-projects-experimental/spring-native/files/6038359/repro-sn507.zip)
See attached. Running mvn package should be enough to generate the error.
Interestingly I have deleted 99% of my project to the point I'm thinking my issue is either environmental or some incredibly obvious config I have wrong.
For reference I'm running in IDEA, adoptopenjdk 15.0.2 on Windows 10

**Comment by sdeleuze** [comment]
OK so based on my test, integration in IDEA does not work correctly when the phase is not defined, so please check d9edcd4932c141db82f329e3fcbb1fb2f7c40afa and bring it back. I have also introduced a [dedicated IDE integration documentation](https://github.com/spring-projects-experimental/spring-native/blob/master/spring-native-docs/src/main/asciidoc/ide-integration.adoc) that should help you.

**Comment by leccelecce** [comment]
@sdeleuze unfortunately this doesn't seem to have helped. I have introduced the phase back in and also tried the after build option in IDEA and then tried rebuilding the project manually in IDEA but still have the exact same error.

I don't think my problem is the order of compilation vs aot generation - looking at the Maven build log compile occurs first then AOT - it more seems like for some reason AOT cannot see the files created in the previous step. For example: if I just run `mvn clean compile spring-aot:generate` I get the exact same error.

**Comment by leccelecce** [comment]
Now debugging spring-aot. It seems the problem may be in TypeSystem - specifically that my application's class files are never added to the index. `appPackages` never gets populated. This might be a forward-slash vs back-slash issue - seems Files.walk is reporting the location as `...\target\classes` but the code is looking for lastIndexOf("/") which isn't found, resulting in the map() returning null.

![image](https://user-images.githubusercontent.com/24962424/109143101-90cac500-7757-11eb-9c1d-065513de4c67.png)

**Comment by leccelecce** [comment]
Fixing the above then leads to another separtor problem in TypeSystem.find(), as the `appPackages` List contains backslashes but slashedTypeName has forwards.

![image](https://user-images.githubusercontent.com/24962424/109144858-a3de9480-7759-11eb-9041-7444c914e1d9.png)

**Comment by sdeleuze** [comment]
Looks like a windows related issue, I will do more test on that plateform.

**Comment by leccelecce** [comment]
I should also point out that the change I made above fixed the generate operation, but test-generate is still failing. Looks like there are some issues in the code that identifies test classes as well.

**Comment by sdeleuze** [comment]
Fixed by 1d54390cd880cc8052ded88a8cd42acad3ccf482

**Comment by sergey-morenets** [comment]
Hi

Was it fixed in 0.9.0?
I have spring-native 0.9.0 and spring-aot-maven-plugin 0.9.0 and issue still reproduces:
`org.springframework.nativex.type.MissingTypeException: Unable to find class file for HelloWorld`
